lib/reversetunnel/conn.go

```diff
@@ -45,6 +45,8 @@ import (
 type connKey struct {
```

Comment:
Was the choice to make scope part of the key done to save effort, or were you thinking that we might have a usecase for the same UUID existing in multiple scopes in the future?

My gut feeling is that I'd rather a scope mismatch at this level be an explicit and unambiguous error rather than a not found, as this kind of mismatch should only ever happen if there are bugs/hijinks in play.  Ex:

```golang
conn, _ := getConn(uuid, connType)
if params.TargetScope != "" && scopes.Compare(params.TargetScope, conn.Scope) != scopes.Equivalent {
    return trace.Errorf("uh-oh!")
}
```

  But I'm also open to leaving as-is if keying by scope is saving significant effort.  This is ultimately a minor distinction I think.